agate/table/from_json.py

```diff
@@ -29,33 +31,58 @@ def from_json(cls, path, row_names=None, key=None, newline=False, column_types=N
         If `True` then the file will be parsed as "newline-delimited JSON".
     :param column_types:
         See :meth:`.Table.__init__`.
+    :param encoding:
+        According to RFC4627, JSON text shall be encoded in Unicode; the default encoding is
+        UTF-8. You can override this by using any encoding supported by your Python's open() function
+        if :code:`path` is a filepath. If passing in a file handle, it is assumed you have already opened it with the correct
+        encoding specified.
     """
     from agate.table import Table
 
     if key is not None and newline:
         raise ValueError('key and newline may not be specified together.')
 
+    close = False
+
+    if six.PY2:
+        kwargs['encoding'] = encoding
+
     if newline:
         js = []
 
         if hasattr(path, 'read'):
             for line in path:
                 js.append(json.loads(line, object_pairs_hook=OrderedDict, parse_float=Decimal, **kwargs))
         else:
-            with open(path, 'r') as f:
-                for line in f:
-                    js.append(json.loads(line, object_pairs_hook=OrderedDict, parse_float=Decimal, **kwargs))
+            if six.PY2:
+                f = open(path, 'Urb')
+            else:
+                f = io.open(path, encoding=encoding)
+
+            close = True
+
+            for line in f:
+                js.append(json.loads(line, object_pairs_hook=OrderedDict, parse_float=Decimal, **kwargs))
     else:
         if hasattr(path, 'read'):
             js = json.load(path, object_pairs_hook=OrderedDict, parse_float=Decimal, **kwargs)
         else:
-            with open(path, 'r') as f:
-                js = json.load(f, object_pairs_hook=OrderedDict, parse_float=Decimal, **kwargs)
+            if six.PY2:
+                f = open(path, 'Urb')
+            else:
+                f = io.open(path, encoding=encoding)
```

Comment:
Why `io.open` and not just `open`?

Reply:
Because `io.open()` is valid for both Python 2 and 3, even with the `encoding` keyword.
- On Python 3, `open()` is an alias for `io.open()`.
- On Python 2, `io.open()` is a backport from Python 3, available from Python 2.6 onwards, and makes working with files easier (see https://docs.python.org/2/library/io.html)